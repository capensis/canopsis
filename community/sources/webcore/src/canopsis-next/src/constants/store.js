export const ENTITIES_TYPES = {
  alarm: 'alarm',
  alarmDetails: 'alarmDetails',
  alarmDetailsSteps: 'alarmDetailsSteps',
  entity: 'entity',
  service: 'entityservices',
  weatherService: 'weatherService',
  pbehavior: 'pbehavior',
  event: 'event',
  userPreference: 'userPreference',
  group: 'group',
  view: 'view',
  viewTab: 'viewTab',
  widget: 'widget',
  widgetTemplate: 'widgetTemplate',
  stat: 'stat',
  user: 'user',
  role: 'role',
  eventFilter: 'eventFilter',
  metaAlarmRule: 'metaAlarmRule',
  snmpRule: 'snmpRule',
  dynamicInfo: 'dynamicInfo',
  broadcastMessage: 'broadcastMessage',
  playlist: 'playlist',
  pbehaviorExceptions: 'pbehaviorExceptions',
  pbehaviorTypes: 'pbehaviorTypes',
  pbehaviorReasons: 'pbehaviorReasons',
  remediationInstruction: 'remediationInstruction',
  remediationJob: 'remediationJob',
  remediationConfiguration: 'remediationConfiguration',
  remediationInstructionExecution: 'remediationInstructionExecution',
  remediationInstructionStats: 'remediationInstructionStats',
  scenario: 'scenario',
  entityCategory: 'entityCategory',
  testSuite: 'testSuite',
  testSuiteHistory: 'testSuiteHistory',
  stateSetting: 'stateSetting',
  idleRules: 'idleRules',
  flappingRules: 'flappingRules',
  resolveRules: 'resolveRules',
  filter: 'filter',
  ratingSettings: 'ratingSettings',
  pattern: 'pattern',
  map: 'map',
  alarmTag: 'alarmTag',
  shareToken: 'shareToken',
  declareTicketRule: 'declareTicketRule',
};
