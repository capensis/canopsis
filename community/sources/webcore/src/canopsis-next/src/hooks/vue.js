import { getCurrentInstance } from 'vue';

/**
 * Hook for get current component instance
 *
 * @return {Vue}
 */
export const useComponentInstance = () => {
  const instance = getCurrentInstance();

  return instance?.proxy || instance;
};

/**
 * Hook for get current component options
 *
 * @return {Object}
 */
export const useComponentOptions = () => {
  const instance = useComponentInstance();

  return instance.$options;
};

/**
 * Hook for get current component model
 *
 * @return {Object}
 */
export const useComponentModel = () => {
  const { model } = useComponentOptions();

  return {
    event: model?.event ?? 'input',
    prop: model?.event ?? 'value',
  };
};
