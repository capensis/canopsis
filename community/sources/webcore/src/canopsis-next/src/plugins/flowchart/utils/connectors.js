import { range } from 'lodash';

import uid from '@/helpers/uid';

/**
 * @typedef {Object} ConnectorPoint
 * @property {number} x
 * @property {number} y
 * @property {string} _id
 */

/**
 * Generate connector
 *
 * @param {number} x
 * @param {number} y
 * @returns {ConnectorPoint}
 */
export const generateConnectorPoint = (x, y) => ({
  x,
  y,
  _id: uid(),
});

/**
 * Generate connector points by rectangle
 *
 * @param {Rect} rect
 * @param {number} count
 * @returns {ConnectorPoint[]}
 */
export const generateConnectorsByRect = (rect, count) => {
  const { width, height, rx, ry, x, y } = rect;
  const sideConnectorsCount = Math.round(count / 4);
  const sideSpacesCount = sideConnectorsCount + 1;
  const startX = x + rx;
  const startY = x + ry;

  const widthConnectorOffset = (width - rx * 2) / (sideSpacesCount);
  const heightConnectorOffset = (height - ry * 2) / (sideSpacesCount);

  const sideConnectors = range(1, sideConnectorsCount + 1);

  const topConnectors = sideConnectors.map(index => generateConnectorPoint(
    startX + widthConnectorOffset * index,
    y,
  ));

  const rightConnectors = sideConnectors.map(index => generateConnectorPoint(
    x + width,
    startY + heightConnectorOffset * index,
  ));

  const bottomConnectors = sideConnectors.map(index => generateConnectorPoint(
    startX + widthConnectorOffset * index,
    y + height,
  ));

  const leftConnectors = sideConnectors.map(index => generateConnectorPoint(
    x,
    startY + heightConnectorOffset * index,
  ));

  const connectors = [
    ...topConnectors,
    ...rightConnectors,
    ...bottomConnectors,
    ...leftConnectors,
  ];

  if (!rx && !ry) {
    /* Angles connectors */
    connectors.push(
      generateConnectorPoint(x, y),
      generateConnectorPoint(x + width, y),
      generateConnectorPoint(x + width, y + height),
      generateConnectorPoint(x, y + height),
    );
  }

  return connectors;
};
