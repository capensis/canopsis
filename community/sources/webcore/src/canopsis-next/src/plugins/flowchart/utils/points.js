/**
 * Calculate points with ghosts points
 *
 * @param {Array} points
 * @returns {Array}
 */
export const getPointsWithGhosts = points => points.reduce((acc, point, index) => {
  const nextIndex = index + 1;
  const nextPoint = points[nextIndex];

  acc.push({ ...point, index });

  if (nextPoint) {
    acc.push({
      x: (point.x + nextPoint.x) / 2,
      y: (point.y + nextPoint.y) / 2,
      ghost: true,
      index: nextIndex,
    });
  }

  return acc;
}, []);
