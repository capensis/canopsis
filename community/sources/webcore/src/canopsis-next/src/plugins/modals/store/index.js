import Vue from 'vue';

import { VUETIFY_ANIMATION_DELAY } from '@/config';

import uid from '@/helpers/uid';

export const types = {
  SHOW: 'SHOW',
  HIDE: 'HIDE',
  HIDE_COMPLETED: 'HIDE_COMPLETED',
  MINIMIZE: 'MINIMIZE',
  MAXIMIZE: 'MAXIMIZE',
};

export default {
  namespaced: true,
  state: {
    allIds: [],
    byId: {},
  },
  getters: {
    modals: state => state.allIds.map(id => state.byId[id]),
    hasMaximizedModal: state => state.allIds.some(id => !state.byId[id].minimized),
  },
  mutations: {
    [types.SHOW](state, {
      id,
      name,
      config = {},
      dialogProps = {},
    }) {
      Vue.set(state.byId, id, {
        id,
        name,
        config,
        dialogProps,
        hidden: false,
        minimized: false,
      });

      state.allIds.push(id);
    },
    [types.HIDE](state, { id }) {
      Vue.set(state.byId[id], 'hidden', true);
    },
    [types.HIDE_COMPLETED](state, { id }) {
      state.allIds = state.allIds.filter(value => value !== id);

      Vue.delete(state.byId, id);
    },
    [types.MINIMIZE](state, { id }) {
      Vue.set(state.byId[id], 'minimized', true);
    },
    [types.MAXIMIZE](state, { id }) {
      Vue.set(state.byId[id], 'minimized', false);
    },
  },
  actions: {
    /**
     * Show modal window by name with unique id
     *
     * @param {Function} commit
     * @param {Object} state
     * @param {string} name
     * @param {Object} [config = {}]
     * @param {Object} [dialogProps = {}]
     * @param {string} [id = uid()]
     */
    show({ commit, state }, {
      name,
      config = {},
      dialogProps = {},
      id = uid('modal'),
    } = {}) {
      if (state.byId[id]) {
        return commit(types.MAXIMIZE, { id });
      }

      return commit(types.SHOW, {
        id,
        name,
        config,
        dialogProps,
      });
    },

    /**
     * Hide modal by id
     *
     * @param {Function} commit
     * @param {Object} state
     * @param {string} [id]
     */
    hide({ commit, state }, { id } = {}) {
      if (!id) {
        throw new Error('Missed required parameter');
      }

      if (!state.byId[id]) {
        return;
      }

      commit(types.HIDE, { id });

      /**
       * This function added for vuetify animation waiting
       */
      setTimeout(() => {
        if (state.byId[id]) {
          commit(types.HIDE_COMPLETED, { id });
        }
      }, VUETIFY_ANIMATION_DELAY);
    },
    /**
     * Minimize modal by id
     *
     * @param {Function} commit
     * @param {string} [id]
     */
    minimize({ commit }, { id } = {}) {
      if (!id) {
        throw new Error('Missed required parameter');
      }

      commit(types.MINIMIZE, { id });
    },

    /**
     * Maximize modal by id
     *
     * @param {Function} commit
     * @param {string} [id]
     */
    maximize({ commit }, { id } = {}) {
      if (!id) {
        throw new Error('Missed required parameter');
      }

      commit(types.MAXIMIZE, { id });
    },
  },
};
