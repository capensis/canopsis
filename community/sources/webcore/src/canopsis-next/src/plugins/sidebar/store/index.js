import { VUETIFY_ANIMATION_DELAY } from '@/config';

export const types = {
  SHOW: 'SHOW',
  HIDE: 'HIDE',
  HIDE_COMPLETED: 'HIDE_COMPLETED',
};

export default {
  namespaced: true,
  state: {
    name: null,
    config: {},
    hidden: true,
  },
  getters: {
    sidebar: state => state,
  },
  mutations: {
    [types.SHOW](state, { name, config = {} }) {
      state.name = name;
      state.config = config;
      state.hidden = false;
    },
    [types.HIDE](state) {
      state.hidden = true;
    },
    [types.HIDE_COMPLETED](state) {
      state.name = null;
      state.config = {};
    },
  },
  actions: {
    show({ commit }, { name, config = {} } = {}) {
      commit(types.SHOW, { name, config });
    },

    hide({ commit, state }) {
      commit(types.HIDE);

      /**
       * This function added for vuetify animation waiting
       */
      setTimeout(() => {
        if (state.hidden) {
          commit(types.HIDE_COMPLETED);
        }
      }, VUETIFY_ANIMATION_DELAY);
    },
  },
};
