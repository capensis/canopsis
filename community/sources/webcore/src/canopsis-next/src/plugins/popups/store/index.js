import { POPUP_TYPES } from '@/constants';
import { POPUP_AUTO_CLOSE_DELAY } from '@/config';

import { uid } from '@/helpers/uid';

export const types = {
  ADD: 'ADD',
  REMOVE: 'REMOVE',
  SET_DEFAULT_CLOSE_TIME: 'SET_DEFAULT_CLOSE_TIME',
};

export default {
  namespaced: true,
  state: {
    defaultCloseTimeByType: {
      [POPUP_TYPES.error]: POPUP_AUTO_CLOSE_DELAY,
      [POPUP_TYPES.success]: POPUP_AUTO_CLOSE_DELAY,
      [POPUP_TYPES.info]: POPUP_AUTO_CLOSE_DELAY,
      [POPUP_TYPES.warning]: POPUP_AUTO_CLOSE_DELAY,
    },
    popups: [],
  },
  getters: {
    popups: state => state.popups,
  },
  mutations: {
    [types.ADD](state, { popup }) {
      state.popups.push(popup);
    },
    [types.REMOVE](state, { id }) {
      state.popups = state.popups.filter(v => v.id !== id);
    },
    [types.SET_DEFAULT_CLOSE_TIME](state, { type, time }) {
      state.defaultCloseTimeByType[type] = time;
    },
  },
  actions: {
    add({ commit, state }, {
      id = uid('popup'),
      type,
      text,
      autoClose = state.defaultCloseTimeByType[type],
    } = {}) {
      commit(types.ADD, {
        popup: {
          id, type, text, autoClose,
        },
      });
    },

    setDefaultCloseTime({ commit }, { type, time = POPUP_AUTO_CLOSE_DELAY }) {
      return commit(types.SET_DEFAULT_CLOSE_TIME, { type, time });
    },

    remove({ commit }, { id }) {
      commit(types.REMOVE, { id });
    },

    success({ dispatch }, popup) {
      return dispatch('add', { ...popup, type: POPUP_TYPES.success });
    },

    info({ dispatch }, popup) {
      return dispatch('add', { ...popup, type: POPUP_TYPES.info });
    },

    warning({ dispatch }, popup) {
      return dispatch('add', { ...popup, type: POPUP_TYPES.warning });
    },

    error({ dispatch }, popup) {
      return dispatch('add', { ...popup, type: POPUP_TYPES.error });
    },
  },
};
