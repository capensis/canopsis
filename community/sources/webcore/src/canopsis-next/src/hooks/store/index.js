import { useComponentInstance } from '../vue';

/**
 * Normalizes a given map into an array of objects with `key` and `value` properties.
 * If the input is an array, each element is treated as both the key and value.
 * If the input is an object, it converts the entries into an array of objects.
 *
 * @param {Object|Array} map - The map to normalize, which can be either an object or an array.
 * @returns {Array.<{key: string, value: string}>} An array of objects with `key` and `value` properties.
 */
const normalizeStoreMap = (map = {}) => (
  Array.isArray(map)
    ? map.map(key => ({ key, value: key }))
    : Object.entries(map).map(([key, value]) => ({ key, value }))
);

/**
 * Retrieves the Vuex store instance from the current Vue component instance.
 * This function must be used within the Vue component's `setup()` method.
 *
 * @returns {Object} The Vuex store instance.
 * @throws {Error} If the function is not used within the `setup()` method.
 *
 * @example
 * // In a Vue component
 * import { useStore } from `./path/to/useStore`;
 *
 * export default {
 *   setup() {
 *     const store = useStore();
 *     // Now you can use the store to commit mutations, dispatch actions, or access state
 *     return {
 *       // Your other setup properties
 *     };
 *   }
 * }
 */
export const useStore = () => {
  const vm = useComponentInstance();

  if (!vm) {
    throw new Error('You must use this function within the "setup()" method');
  }

  return vm.$store;
};

/**
 * Creates hooks for accessing Vuex store module's getters and actions.
 * This function helps in creating a more modular and organized way to interact with Vuex store modules.
 *
 * @param {string} namespace - The namespace of the Vuex store module.
 * @returns {Object} An object containing the store, module, and functions to access getters and actions.
 * @throws {Error} If the provided namespace does not correspond to any module in the store.
 *
 * @example
 * // Assuming a Vuex store module is registered under `user/` namespace
 * import { useStoreModuleHooks } from `./path/to/useStoreModuleHooks`;
 *
 * export default {
 *   setup() {
 *     const { useGetters, useActions } = useStoreModuleHooks('user/');
 *     const { userName } = useGetters(['userName']);
 *     const { updateUser } = useActions({ customUpdateUser: 'updateUser' });
 *
 *     return {
 *       userName,
 *       customUpdateUser
 *     };
 *   }
 * }
 */
export const useStoreModuleHooks = (namespace) => {
  const store = useStore();
  const preparedNamespace = namespace.slice(-1) === '/' ? namespace : `${namespace}/`;
  const module = store._modulesNamespaceMap[preparedNamespace];

  if (!module) {
    throw new Error(`Incorrect module namespace = ${namespace}`);
  }

  const useGetters = getters => normalizeStoreMap(getters).reduce((acc, { key, value }) => {
    acc[key] = () => store.getters(preparedNamespace + value);

    return acc;
  }, {});

  const useActions = actions => normalizeStoreMap(actions).reduce((acc, { key, value }) => {
    acc[key] = (...args) => module.context.dispatch.apply(store, [value, ...args]);

    return acc;
  }, {});

  return {
    store,
    module,
    useGetters,
    useActions,
  };
};
