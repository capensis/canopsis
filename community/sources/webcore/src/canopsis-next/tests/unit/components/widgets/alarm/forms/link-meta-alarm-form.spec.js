import Faker from 'faker';

import { flushPromises, generateShallowRenderer, generateRenderer } from '@unit/utils/vue';
import { createInputStub } from '@unit/stubs/input';
import { createMockedStoreModules, createMetaAlarmModule } from '@unit/utils/store';

import LinkMetaAlarmForm from '@/components/widgets/alarm/forms/link-meta-alarm-form.vue';

const stubs = {
  'v-combobox': createInputStub('v-combobox'),
  'v-text-field': createInputStub('v-text-field'),
  'c-enabled-field': true,
};

const snapshotStubs = {
  'c-enabled-field': true,
};

const selectTextField = wrapper => wrapper.find('.v-text-field');
const selectComboboxField = wrapper => wrapper.find('.v-combobox');
const selectEnabledField = wrapper => wrapper.find('c-enabled-field-stub');

describe('link-meta-alarm-form', () => {
  const { metaAlarmModule, fetchMetaAlarmsListWithoutStore } = createMetaAlarmModule();
  const store = createMockedStoreModules([
    metaAlarmModule,
  ]);

  const factory = generateShallowRenderer(LinkMetaAlarmForm, { stubs });
  const snapshotFactory = generateRenderer(LinkMetaAlarmForm, { stubs: snapshotStubs });

  test('Alarms fetched after mount', () => {
    factory({
      store,
      propsData: {
        form: {},
      },
    });

    expect(fetchMetaAlarmsListWithoutStore).toBeCalled();
  });

  test('Meta alarm changed after trigger text field', () => {
    const form = {
      metaAlarm: Faker.datatype.string(),
      comment: Faker.datatype.string(),
      auto_resolve: false,
    };
    const wrapper = factory({
      store,
      propsData: {
        form,
      },
    });

    const metaAlarm = Faker.datatype.string();

    const comboboxField = selectComboboxField(wrapper);

    comboboxField.setValue(metaAlarm);

    expect(wrapper).toEmitInput({ ...form, metaAlarm });
  });

  test('Comment changed after trigger description field', () => {
    const form = {
      metaAlarm: Faker.datatype.string(),
      comment: Faker.datatype.string(),
      auto_resolve: false,
    };
    const wrapper = factory({
      store,
      propsData: {
        form,
      },
    });

    const comment = Faker.datatype.string();

    selectTextField(wrapper).setValue(comment);

    expect(wrapper).toEmitInput({ ...form, comment });
  });

  test('Auto resolve changed after trigger enabled field', () => {
    const form = {
      metaAlarm: Faker.datatype.string(),
      comment: Faker.datatype.string(),
      auto_resolve: false,
    };
    const wrapper = factory({
      store,
      propsData: {
        form,
      },
    });

    const autoResolve = true;

    const enabledField = selectEnabledField(wrapper);

    enabledField.triggerCustomEvent('input', autoResolve);

    expect(wrapper).toEmitInput({ ...form, auto_resolve: autoResolve });
  });

  test('Renders `link-meta-alarm-form` with default props', async () => {
    const wrapper = snapshotFactory({
      store,
      propsData: {
        form: {
          metaAlarm: 'metaAlarm',
          comment: 'comment',
          auto_resolve: true,
        },
      },
    });

    expect(wrapper).toMatchSnapshot();
    await wrapper.activateAllMenus();
    expect(wrapper).toMatchMenuSnapshot();
  });

  test('Renders `link-meta-alarm-form` with alarms', async () => {
    fetchMetaAlarmsListWithoutStore.mockReturnValueOnce([
      { _id: 'entity-id', name: 'alarm-display-name' },
    ]);
    const wrapper = snapshotFactory({
      store: createMockedStoreModules([metaAlarmModule]),
      propsData: {
        form: {},
      },
    });

    await flushPromises();

    expect(wrapper).toMatchSnapshot();
    await wrapper.activateAllMenus();
    expect(wrapper).toMatchMenuSnapshot();
  });
});
