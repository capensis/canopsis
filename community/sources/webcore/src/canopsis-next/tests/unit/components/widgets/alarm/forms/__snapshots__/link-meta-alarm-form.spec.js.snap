// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link-meta-alarm-form Renders \`link-meta-alarm-form\` with alarms 1`] = `
<div
  class="layout column"
>
  <div
    class="v-input theme--light v-text-field v-select v-autocomplete"
  >
    <div
      class="v-input__control"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="v-input__slot"
        role="combobox"
      >
        <div
          class="v-select__slot"
        >
          <label
            class="v-label theme--light"
            style="left: 0px; position: absolute;"
          >
            Select meta alarm or create a new one
          </label>
          <input
            autocomplete="off"
            type="text"
          />
          <div
            class="v-input__append-inner"
          >
            <div
              class="v-input__icon v-input__icon--append"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate material-icons theme--light"
              >
                arrow_drop_down
              </i>
            </div>
          </div>
          <input
            name="manualMetaAlarm"
            type="hidden"
          />
        </div>
        <div
          class="v-menu"
        >
          <!---->
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="v-input theme--light v-text-field"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <label
            class="v-label theme--light"
            style="left: 0px; position: absolute;"
          >
            Note
          </label>
          <input
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
  <c-enabled-field-stub
    label="Auto resolve"
  />
</div>
`;

exports[`link-meta-alarm-form Renders \`link-meta-alarm-form\` with alarms 2`] = `
<div
  class="v-menu__content theme--light menuable__content__active v-autocomplete__content"
  style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 8;"
>
  <div
    class="v-list v-select-list v-sheet theme--light theme--light"
    role="listbox"
    tabindex="-1"
  >
    <div
      aria-selected="false"
      class="v-list-item v-list-item--link theme--light"
      role="option"
      tabindex="0"
    >
      <div
        class="v-list-item__content"
      >
        <div
          class="v-list-item__title"
        >
          alarm-display-name
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`link-meta-alarm-form Renders \`link-meta-alarm-form\` with default props 1`] = `
<div
  class="layout column"
>
  <div
    class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select v-autocomplete"
  >
    <div
      class="v-input__control"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="v-input__slot"
        role="combobox"
      >
        <div
          class="v-select__slot"
        >
          <label
            class="v-label v-label--active theme--light"
            style="left: 0px; position: absolute;"
          >
            Select meta alarm or create a new one
          </label>
          <input
            autocomplete="off"
            type="text"
          />
          <div
            class="v-input__append-inner"
          >
            <div
              class="v-input__icon v-input__icon--append"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate material-icons theme--light"
              >
                arrow_drop_down
              </i>
            </div>
          </div>
          <input
            name="manualMetaAlarm"
            type="hidden"
            value="metaAlarm"
          />
        </div>
        <div
          class="v-menu"
        >
          <!---->
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <label
            class="v-label v-label--active theme--light"
            style="left: 0px; position: absolute;"
          >
            Note
          </label>
          <input
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
  <c-enabled-field-stub
    label="Auto resolve"
    value="true"
  />
</div>
`;

exports[`link-meta-alarm-form Renders \`link-meta-alarm-form\` with default props 2`] = `undefined`;
